iceoryx_posh/test/moduletests/test_popo_client_port.cpp

```diff
@@ -0,0 +1,552 @@
+            constexpr bool UNREACHABLE{false};
+            EXPECT_TRUE(UNREACHABLE);
+        });
+
+    sut.connect();
+    tryAdvanceToState(clientPortDataWithoutConnectOnCreate, iox::ConnectionState::CONNECTED);
+
+    EXPECT_FALSE(serverRequestQueue.tryPop().has_value());
+}
+
+TEST_F(ClientPort_test, GetResponseOnNotConnectedClientPortHasNoResponse)
+{
+    auto& sut = clientPortUserWithoutConnectOnCreate;
+    sut.getResponse()
+        .and_then([&](auto&) {
+            constexpr bool UNREACHABLE{false};
+            EXPECT_TRUE(UNREACHABLE);
+        })
+        .or_else([&](auto& err) { EXPECT_THAT(err, Eq(iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)); });
+}
+
+TEST_F(ClientPort_test, GetResponseOnConnectedClientPortWithEmptyResponseQueueHasNoResponse)
+{
+    auto& sut = clientPortUserWithConnectOnCreate;
+    sut.getResponse()
+        .and_then([&](auto&) {
+            constexpr bool UNREACHABLE{false};
+            EXPECT_TRUE(UNREACHABLE);
+        })
+        .or_else([&](auto& err) { EXPECT_THAT(err, Eq(iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)); });
+}
+
+TEST_F(ClientPort_test, GetResponseOnConnectedClientPortWithNonEmptyResponseQueueHasResponse)
+{
+    constexpr int64_t SEQUENCE_ID{13U};
+    auto& sut = clientPortUserWithConnectOnCreate;
+
+    constexpr uint32_t USER_PAYLOAD_SIZE{10};
+    auto sharedChunk = getChunkFromMemoryManager(USER_PAYLOAD_SIZE, sizeof(ResponseHeader));
+    new (sharedChunk.getChunkHeader()->userHeader())
+        ResponseHeader(iox::UniquePortId(), RpcBaseHeader::UNKNOWN_CLIENT_QUEUE_INDEX, SEQUENCE_ID);
+    chunkQueuePusherWithConnectOnCreate.push(sharedChunk);
+
+    sut.getResponse()
+        .and_then([&](auto& responseHeader) { EXPECT_THAT(responseHeader->getSequenceId(), Eq(SEQUENCE_ID)); })
+        .or_else([&](auto&) {
+            constexpr bool UNREACHABLE{false};
+            EXPECT_TRUE(UNREACHABLE);
+        });
+}
+
+TEST_F(ClientPort_test, ReleaseResponseWithNullptrIsTerminating)
+{
+    auto& sut = clientPortUserWithConnectOnCreate;
+
+    iox::cxx::optional<iox::Error> detectedError;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+            detectedError.emplace(error);
+            EXPECT_EQ(errorLevel, iox::ErrorLevel::SEVERE);
+        });
+
+    sut.releaseResponse(nullptr);
+
+    ASSERT_TRUE(detectedError.has_value());
+    EXPECT_EQ(detectedError.value(), iox::Error::kPOPO__CLIENT_PORT_INVALID_RESPONSE_TO_RELEASE_FROM_USER);
+}
+
+TEST_F(ClientPort_test, ReleaseResponseWithValidResponseReleasesChunkToTheMempool)
+{
+    auto& sut = clientPortUserWithConnectOnCreate;
+
+    constexpr uint32_t USER_PAYLOAD_SIZE{10};
+
+    iox::cxx::optional<iox::mepoo::SharedChunk> sharedChunk{
+        getChunkFromMemoryManager(USER_PAYLOAD_SIZE, sizeof(ResponseHeader))};
+    chunkQueuePusherWithConnectOnCreate.push(sharedChunk.value());
```

Comment:
The value will always be there. The optional is just used to be able to explicitly release the SharedChunk instead of implicitly by creating a new scope. I would also just terminate in `getChunkFromMemoryManager` since this is nothing that can fail in this test.

Reply:
Just to be clear. With terminate I mean a `cxx::Expects` or similar.

Reply:
Added a `cxx::Expects`. Please resolve if this is okay for you